hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -50,6 +62,25 @@ data ClientRequest tx
   | NewTx tx
   | Close
   | Contest
+  deriving (Generic)
+
+instance (ToJSON tx, ToJSON (UTxO tx)) => ToJSON (ClientRequest tx) where
+  toJSON = genericToJSON clientRequestJsonOptions
+
+instance (FromJSON tx, FromJSON (UTxO tx)) => FromJSON (ClientRequest tx) where
+  parseJSON = genericParseJSON clientRequestJsonOptions
+
+clientRequestJsonOptions :: Aeson.Options
+clientRequestJsonOptions =
+  Aeson.defaultOptions
+    { constructorTagModifier = camelCase
```

Comment:
This `camelCase` function is basically `mapHead toLower` and I know now why you were suggesting to avoid `UTxO` as a name as it would turn out as `uTxO` using this constructor modifier :joy:

Not sure though whether we want to have another dependency just for this very simple modifier

Reply:
Instead of adding dependencies and doing fancy meta-programming I'd be even open to hand-written instances. Sometimes they are easier to read and maintain than `genericXXX` + various `options` and `modifiers`

Reply:
> Not sure though whether we want to have another dependency just for this very simple modifier

I initially wrote them by hand, and ended up writing the same one in two places. Wanted to move in a separate module but it felt a bit _too excessive_ / inappropriate. In the end, aeson-casing is a tiny library with almost no dependency (base and aeson) which kinda does just is needed. 

Writing JSON by hand is always doable, just a bit annoying. I think my favorite approach remains generics with default options, and possibly `DuplicateRecordFields` when really necessary. With `NamedFieldPuns` it's usually not a big problem.